packages/firestore/test/unit/remote/remote_event.test.ts

```diff
@@ -460,4 +464,101 @@ describe('RemoteEvent', () => {
     const mapping1 = updateMapping([doc1, doc2], []);
     expectEqual(event.targetChanges[1].mapping, mapping1);
   });
+
+  it('synthesizes deletes', () => {
+    const targets = listens(1, 2, 3);
+    const shouldSynthesize = new WatchTargetChange(
+      WatchTargetChangeState.Current,
+      [1]
+    );
+    const wrongState = new WatchTargetChange(WatchTargetChangeState.NoChange, [
+      2
+    ]);
+    const hasDocument = new WatchTargetChange(WatchTargetChangeState.Current, [
+      3
+    ]);
+    const doc1 = doc('docs/1', 1, { value: 1 });
+    const docChange = new DocumentWatchChange([3], [], doc1.key, doc1);
+
+    const event = remoteEvent(
+      1,
+      targets,
+      noPendingResponses,
+      shouldSynthesize,
+      wrongState,
+      hasDocument,
+      docChange
+    );
+
+    const synthesized = DocumentKey.fromPathString('docs/2');
+    expect(event.documentUpdates.get(synthesized)).to.equal(null);
+
+    const limboTargetChange = event.targetChanges[1];
+    event.synthesizeDeleteForLimboTargetChange(limboTargetChange, synthesized);
+    const expected = deletedDoc(
+      'docs/2',
+      event.snapshotVersion.toMicroseconds()
+    );
+    expectEqual(expected, event.documentUpdates.get(synthesized));
+
+    const notSynthesized = DocumentKey.fromPathString('docs/no1');
+    event.synthesizeDeleteForLimboTargetChange(
+      event.targetChanges[2],
```

Comment:
Do you mind adding the little helper variables here and below like you did for `limboTargetChange`?